pypesto/sampling/metropolis.py

```diff
@@ -29,31 +31,41 @@ def default_options(cls):
 
     def initialize(self, problem: Problem, x0: np.ndarray):
         self.problem = problem
-        self.objective = problem.objective
-        self.objective.history = History()
+        self.neglogpost = problem.objective
+        self.neglogpost.history = History()
+        if problem.x_priors is None:
+            self.neglogprior = lambda x: -0.
+        else:
+            self.neglogprior = problem.x_priors
         self.trace_x = [x0]
-        self.trace_fval = [self.objective(x0)]
+        self.trace_neglogpost = [self.neglogpost(x0)]
+        self.trace_neglogprior = [self.neglogprior(x0)]
 
-    def sample(self, n_samples: int, beta: float = 1.):
+    def sample(self, n_samples: int, beta: float = 1.,
+               temper_lpost: bool = False):
```

Comment:
can't add `temper_lpost` ad an argument to the sample function. please define it in `__init__`.

Reply:
TODO